[rocblas][AdaptiveCpp] Build issues with missing pi headers

While building with AdaptiveCPP compiler (v25.02.0), most domains would likely stumble into this missing header. Upon looking at the preprocessor-macro setup in the concerned header, it doesnt handle the case for AdaptiveCPP. I wasn't sure if this is compiler is being actively tracked.

Build cmake:
```
cmake .. \
      -DCMAKE_C_COMPILER=acpp \
      -DCMAKE_CXX_COMPILER=acpp \
      -DONEMATH_SYCL_IMPLEMENTATION=hipsycl \
      -DAdaptiveCpp_DIR=${ACPP_HOME}/acpp/lib/cmake/AdaptiveCpp \
      -DENABLE_MKLGPU_BACKEND=False \
      -DENABLE_MKLCPU_BACKEND=False \
      -DENABLE_ROCBLAS_BACKEND=True \
      -DENABLE_ROCSOLVER_BACKEND=True \
      -DENABLE_ROCRAND_BACKEND=True \
      -DENABLE_ROCSPARSE_BACKEND=False \
      -DHIPSYCL_TARGETS=hip:gfx90a \
      -DBUILD_FUNCTIONAL_TESTS=False \
      -DBUILD_EXAMPLES=False \
      -DHIP_TARGETS=gfx90a \
      -DCMAKE_INSTALL_PREFIX=${ACPP_HOME}/onemath
```
Compiler Version: (AdaptiveCPP:v25.02.0)
```bash
acpp --version
AMD clang version 19.0.0git (https://github.com/RadeonOpenCompute/llvm-project roc-6.4.0 25133 c7fe45cf4b819c5991fe208aaa96edf142730f1d)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /opt/rocm-6.4.0/lib/llvm/bin
Configuration file: /opt/rocm-6.4.0/lib/llvm/bin/clang++.cfg
```
Error
```bash
[ 30%] Building CXX object bin/blas/backends/rocblas/CMakeFiles/onemath_blas_rocblas_obj.dir/rocblas_level1.cpp.o
cd /lustre/orion/csc613/scratch/abagusetty/SYCL_compilers_05282025/oneMath/build-acpp-rocm6.4.0/bin/blas/backends/rocblas &&  /lustre/orion/csc613/scratch/abagusetty/SYCL_compilers_05282025/acpp_install-6.4.0_05282025/acpp/lib/cmake/hipSYCL/syclcc-launcher --launcher-cxx-compiler=/lustre/orion/csc613/scratch/abagusetty/SYCL_compilers_05282025/acpp_install-6.4.0_05282025/acpp/bin/acpp --launcher-syclcc=/lustre/orion/csc613/scratch/abagusetty/SYCL_compilers_05282025/acpp_install-6.4.0_05282025/acpp/bin/syclcc-clang  --hipsycl-targets="hip:gfx90a" /lustre/orion/csc613/scratch/abagusetty/SYCL_compilers_05282025/acpp_install-6.4.0_05282025/acpp/bin/acpp -DUSE_PROF_API=1 -D__HIP_PLATFORM_AMD__=1 -I/lustre/orion/csc613/scratch/abagusetty/SYCL_compilers_05282025/oneMath/include -I/lustre/orion/csc613/scratch/abagusetty/SYCL_compilers_05282025/oneMath/build-acpp-rocm6.4.0/bin -I/lustre/orion/csc613/scratch/abagusetty/SYCL_compilers_05282025/oneMath/src/include -I/lustre/orion/csc613/scratch/abagusetty/SYCL_compilers_05282025/oneMath/src -isystem /opt/rocm-6.4.0/include -isystem /lustre/orion/csc613/scratch/abagusetty/SYCL_compilers_05282025/acpp_install-6.4.0_05282025/acpp/include -O3 -DNDEBUG -std=gnu++17 -fPIC -MD -MT bin/blas/backends/rocblas/CMakeFiles/onemath_blas_rocblas_obj.dir/rocblas_level1.cpp.o -MF CMakeFiles/onemath_blas_rocblas_obj.dir/rocblas_level1.cpp.o.d -o CMakeFiles/onemath_blas_rocblas_obj.dir/rocblas_level1.cpp.o -c /lustre/orion/csc613/scratch/abagusetty/SYCL_compilers_05282025/oneMath/src/blas/backends/rocblas/rocblas_level1.cpp
acpp warning: syclcc-clang is deprecated; please use acpp instead.
In file included from /lustre/orion/csc613/scratch/abagusetty/SYCL_compilers_05282025/oneMath/src/blas/backends/rocblas/rocblas_level1.cpp:23:
/lustre/orion/csc613/scratch/abagusetty/SYCL_compilers_05282025/oneMath/src/blas/backends/rocblas/rocblas_task.hpp:46:10: fatal error: 'CL/sycl/detail/pi.hpp' file not found
   46 | #include <CL/sycl/detail/pi.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~
1 error generated when compiling for gfx90a.

```

--- a/src/blas/backends/rocblas/rocblas_scope_handle.hpp
+++ b/src/blas/backends/rocblas/rocblas_scope_handle.hpp
@@ -19,7 +19,11 @@
 **************************************************************************/
 #ifndef _ROCBLAS_SCOPED_HANDLE_HPP_
 #define _ROCBLAS_SCOPED_HANDLE_HPP_
+#if __has_include(<sycl/sycl.hpp>)
+#include <sycl/sycl.hpp>
+#else
 #include <CL/sycl.hpp>
+#endif
 #include <memory>
 #include <thread>
 #include <atomic>

--- a/src/blas/backends/rocblas/rocblas_task.hpp
+++ b/src/blas/backends/rocblas/rocblas_task.hpp
@@ -34,18 +34,6 @@
 #include "rocblas_scope_handle_hipsycl.hpp"
 #endif
 
-// After Plugin Interface removal in DPC++ ur.hpp is the new include
-#if __has_include(<sycl/detail/ur.hpp>)
-#include <sycl/detail/ur.hpp>
-#ifndef ONEMATH_PI_INTERFACE_REMOVED
-#define ONEMATH_PI_INTERFACE_REMOVED
-#endif
-#elif __has_include(<sycl/detail/pi.hpp>)
-#include <sycl/detail/pi.hpp>
-#else
-#include <CL/sycl/detail/pi.hpp>
-#endif
-
 namespace oneapi {
 namespace math {
 namespace blas {

